### PR TITLE
docs: ✏️ Add non-cli section for new zip-upload functionality

### PR DESCRIPTION
**Pull request by richard-wpengine**
Added non-cli section, but under cli-resources, in order to host the new non-CLI (CURL) zip-upload functionality that allows upload-enabled accounts to modify existing Atlas environments by updating them from a pushed zip of local repo. This is agnostic of CLI but requires existing applicable account, app & env, local repo, curl, and the generation of API credentials in portal

✅ Closes: GAIA-2892

### CODE REVIEW

File: atlas/reference/non-cli/zip-upload.mdx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+```bash
+curl -X POST -H "Content-Type: multipart/form-data" -F "repository=@PATH-TO-FILE.zip" -u "PUT_API_USERNAME_HERE:PUT_API_PASSWORD_HERE" https://js.wpengineapi.com/fu/v1/accounts/{PUT_ACCOUNT_NAME_HERE}/apps/{PUT_APP_NAME_HERE}/environments/{PUT_ENVIRONMENT_UUID}:upload
+```
+The following is what the CURL command with populated credentials should look like:
+```bash
+curl -X POST -H "Content-Type: multipart/form-data" -F "repository=@express-simple-server-main.zip" -u "user_name:password" https://js.wpengineapi.com/fu/v1/accounts/wpengineaccount/apps/faust-app-001/environments/135788ee-e49f-0670-1a83-9c4eeede01f9:upload
+```
+
+###Notes:
+CURL HTTP Response will give `200` status, but will not return a response body, if successful. If you would like more detail add the `-i` (light summary) or `-v` (detailed summary) flag to the avove CURL command. You can also check Portal to see the Atlas Environment rebuilding state.
```
Comment (klemis) [inline atlas/reference/non-cli/zip-upload.mdx:28]:
```suggestion
CURL HTTP Response will give `200` status, but will not return a response body, if successful. If you would like more detail add the `-i` (include the HTTP response headers in the output) or `-v` (makes curl verbose during the operation) flag to the above CURL command. You can also check Portal to see the Atlas Environment rebuilding state.
```

Comment (klemis) [reply atlas/reference/non-cli/zip-upload.mdx:28]: How about a slightly more detailed description of the flags?

Comment (richard-wpengine) [reply atlas/reference/non-cli/zip-upload.mdx:28]: Any suggestion on the wording?

Comment (klemis) [reply atlas/reference/non-cli/zip-upload.mdx:28]: I meant the ones mentioned above 😄

Comment (richard-wpengine) [reply atlas/reference/non-cli/zip-upload.mdx:28]: Ah, my bad. 👯‍♂️

**Review by klemis — APPROVED** [review]
LGTM!

**Review by Fran-A-Dev — APPROVED** [review]
LGTM

**Pull request merged**